main.go

```diff
@@ -6,131 +22,81 @@ import (
-	klog.InitFlags(nil)
-	flag.Set("v", "2")
-
-	// Add the zap logger flag set to the CLI. The flag set must
-	// be added before calling pflag.Parse().
-	pflag.CommandLine.AddFlagSet(zap.FlagSet())
-	pflag.StringVar(&logType, "v", "production", "Log type (debug/production).")
-	// Add flags registered by imported packages (e.g. glog and
-	// controller-runtime)
-	pflag.CommandLine.AddGoFlagSet(flag.CommandLine)
-
-	pflag.Parse()
-
-	// Use a zap logr.Logger implementation. If none of the zap
-	// flags are configured (or if the zap flag set is not being
-	// used), this defaults to a production zap logger.
-	//
-	// The logger instantiated here can be changed to any logger
-	// implementing the logr.Logger interface. This logger will
-	// be propagated through the whole operator, generating
-	// uniform and structured logs.
-	if logType == "debug" {
-		logf.SetLogger(logf.ZapLogger(true))
-	} else {
-		logf.SetLogger(logf.ZapLogger(false))
-	}
+	var metricsAddr string
+	flag.StringVar(&metricsAddr, "metrics-addr", ":8080", "The address the metric endpoint binds to.")
+	flag.Parse()
 
-	printVersion()
+	ctrl.SetLogger(zap.New(zap.UseDevMode(true)))
 
 	// Lock only for handler, we can run old and new version of
 	// webhook without problems, policy status will be updated
 	// by multiple instances.
 	if environment.IsHandler() {
 		handlerLock, err := lockHandler()
 		if err != nil {
-			log.Error(err, "Failed to run lockHandler")
+			setupLog.Error(err, "Failed to run lockHandler")
 			os.Exit(1)
 		}
 		defer handlerLock.Unlock()
-		log.Info("Successfully took nmstate exclusive lock")
+		setupLog.Info("Successfully took nmstate exclusive lock")
 	}
 
-	namespace, err := k8sutil.GetWatchNamespace()
-	if err != nil {
-		log.Error(err, "Failed to get watch namespace")
-		os.Exit(1)
-	}
-
-	// Get a config to talk to the apiserver
-	cfg, err := config.GetConfig()
-	if err != nil {
-		log.Error(err, "")
-		os.Exit(1)
-	}
-
-	mgrOptions := manager.Options{
-		Namespace:      namespace,
-		MapperProvider: apiutil.NewDiscoveryRESTMapper,
+	ctrlOptions := ctrl.Options{
+		Scheme: scheme,
 	}
 
 	// We need to add LeaerElection for the webhook
-	// cert-manager
+	// cert-manager the LeaderElectionID was generated by operator-sdk
 	if environment.IsWebhook() {
-		mgrOptions.LeaderElection = true
-		mgrOptions.LeaderElectionID = "nmstate-webhook-lock"
-		mgrOptions.LeaderElectionNamespace = namespace
+		ctrlOptions.LeaderElection = true
+		ctrlOptions.LeaderElectionID = "5d2e944a.nmstate.io"
 	}
 
-	// Create a new Cmd to provide shared dependencies and start components
-	mgr, err := manager.New(cfg, mgrOptions)
+	mgr, err := ctrl.NewManager(ctrl.GetConfigOrDie(), ctrlOptions)
 	if err != nil {
-		log.Error(err, "")
-		os.Exit(1)
-	}
-
-	log.Info("Registering Components.")
-
-	// Setup Scheme for all resources
-	if err := api.AddToScheme(mgr.GetScheme()); err != nil {
-		log.Error(err, "")
+		setupLog.Error(err, "unable to start manager")
```

Comment:
It does not start it, just create it. No?

Reply:
I am with you, don't know why they generate like this... Do we keep like the one emitted or do we change it ?

Reply:
Puff. Let's keep it then and later thing how to unify it. Either propose a PR to kubebuilder or to adjust all our other logs.